src/lib.rs

```diff
@@ -132,14 +121,15 @@ impl CredentialHandler {
         // handle this we bail out of this authentication session after setting
         // the flag `ssh_username_requested`, and then we handle this below.
         if allowed.contains(git2::CredentialType::USERNAME) {
-            // debug_assert!(username.is_none());
+            debug_assert!(username.is_none());
```

Comment:
I don't want the lib to panic if `username` is defined. Can you comment the statement again or handle this case differently?

Reply:
Thank you for reviewing!

fixed by a2ee3078bcf63f806dfb639902727f4599bb5fd3